t/unit-tests/u-example-decorate.c

```diff
@@ -0,0 +1,104 @@
+#define USE_THE_REPOSITORY_VARIABLE
+
+#include "unit-test.h"
+#include "object.h"
+#include "decorate.h"
+#include "repository.h"
+
+struct test_vars {
+	struct object *one, *two, *three;
+	struct decoration n;
+	int decoration_a, decoration_b;
+};
+
+static void t_add(struct test_vars *vars)
+{
+	void *ret = add_decoration(&vars->n, vars->one, &vars->decoration_a);
+
+	cl_assert(ret == NULL);
+	ret = add_decoration(&vars->n, vars->two, NULL);
+	cl_assert(ret == NULL);
+}
+
+static void t_readd(struct test_vars *vars)
+{
+	void *ret = add_decoration(&vars->n, vars->one, NULL);
+
+	cl_assert(ret == &vars->decoration_a);
+	ret = add_decoration(&vars->n, vars->two, &vars->decoration_b);
+	cl_assert(ret == NULL);
+}
+
+static void t_lookup(struct test_vars *vars)
+{
+	void *ret = lookup_decoration(&vars->n, vars->one);
+
+	cl_assert(ret == NULL);
+	ret = lookup_decoration(&vars->n, vars->two);
+	cl_assert(ret == &vars->decoration_b);
+	ret = lookup_decoration(&vars->n, vars->three);
+	cl_assert(ret == NULL);
+}
+
+static void t_loop(struct test_vars *vars)
+{
+	int objects_noticed = 0;
+
+	for (size_t i = 0; i < vars->n.size; i++) {
+		if (vars->n.entries[i].base)
+			objects_noticed++;
+	}
+	cl_assert_equal_i(objects_noticed, 2);
+}
+
+void test_example_decorate__add(void)
+{
+	struct object_id one_oid = { { 1 } }, two_oid = { { 2 } }, three_oid = { { 3 } };
+	struct test_vars vars = { 0 };
+
+	vars.one = lookup_unknown_object(the_repository, &one_oid);
+	vars.two = lookup_unknown_object(the_repository, &two_oid);
+	vars.three = lookup_unknown_object(the_repository, &three_oid);
+
+	t_add(&vars);
+	clear_decoration(&vars.n, NULL);
+}
+
+void test_example_decorate__readd(void)
+{
+	struct object_id one_oid = { { 1 } }, two_oid = { { 2 } }, three_oid = { { 3 } };
+	struct test_vars vars = { 0 };
+
+	vars.one = lookup_unknown_object(the_repository, &one_oid);
+	vars.two = lookup_unknown_object(the_repository, &two_oid);
+	vars.three = lookup_unknown_object(the_repository, &three_oid);
```

Comment:
This setup code is repeated across all tests. You could instead introduce a function `void test_exmaple_decorate__init(void)` that initializes a global `vars` variable.